src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'rectangle' | 'circle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    a: number,
+    b: number,
+    c: number,
+    public shape: 'triangle',
+    public color: 'red' | 'green' | 'blue',
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Length is not allowed to be lower than zero');
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('Measurements do not match');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiPerimeter *
+        (semiPerimeter - this.a) *
+        (semiPerimeter - this.b) *
+        (semiPerimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  radius: number;
 
-export class Circle implements Figure {}
+  constructor(
+    radius: number,
+    public shape: 'circle',
+    public color: 'red' | 'green' | 'blue',
+  ) {
+    this.radius = radius;
 
-export class Rectangle implements Figure {}
+    if (this.radius <= 0) {
+      throw new Error('Radius is not allowed to be lower than zero');
```

Comment:
[CODE STYLE] - The error message here could be improved for clarity. Instead of 'Radius is not allowed to be lower than zero', you might say 'Radius must be greater than zero'.